CleanWeatherTests/Scenes/CityList/CityListWorkerTests.swift

```diff
@@ -84,45 +84,52 @@ final class CityListWorkerTests: QuickSpec {
 
         }
 
-        describe("fetching cities") {
+        describe("fetching favourite cities weather") {
 
-            var receivedCities: [CityWeather]?
+            var receivedCitiesWeather: [CityWeather]?
             var receivedError: Error?
 
             beforeEach {
-                worker.fetchCityWeather { result in
+                worker.fetchCitiesWeather(cities: Mock.favouriteCities){ result in
                     switch result {
                     case .success(let cities):
-                        receivedCities = cities
+                        receivedCitiesWeather = cities
+                        //FIXME: Why I got Tokyo 4 times?
+                        print("HERE-->", receivedCitiesWeather, "<--HERE")
```

Comment:
you get tokyo 4 times, because in worker Tokyo is the last city, and everytime it calls networking, our fake networking is replacing it's private completion block. So you actually call last completion 4 times instead of all only once


you'd need to fix fakenetworking for this case, by having completion block in a kind of array

    var fetchCurrentWeatherForCityCompletion: [FetchWeatherResponseCompletion]?

but you'd need to clean it and also verify if count is 4

Reply:
It's Alive!!!
But why I have to clean it and verify if count is 4? It works without it.
What am I missing?